modules/nixpkgs.nix

```diff
@@ -11,14 +11,195 @@
+          options = {
+            config = mkOption {
+              default = { };
+              description = mdDoc ''
+                Config for this Nixpkgs evaluation.
+              '';
+              type = submoduleWithModules { } {
+                _file = ./nixpkgs.nix;
+                freeformType = types.lazyAttrsOf types.raw;
+              };
+            };
+            crossOverlays = mkOption {
+              default = [ ];
+              description = mdDoc ''
+                List of Nixpkgs overlays to apply to target packages only for this Nixpkgs evaluation.
+              '';
+              type = types.listOf (types.uniq (types.functionTo (types.functionTo (types.lazyAttrsOf types.unspecified))));
+            };
+            overlays = mkOption {
+              default = [ ];
+              description = mdDoc ''
+                List of Nixpkgs overlays for this Nixpkgs evaluation.
+              '';
+              type = types.listOf (types.uniq (types.functionTo (types.functionTo (types.lazyAttrsOf types.unspecified))));
+            };
+          };
+        });
+      };
+    };
+    perSystem = mkPerSystemOption ({ config, system, ... }: {
+      _file = ./nixpkgs.nix;
+      options = {
+        nixpkgs = {
+          evals = mkOption {
+            default = { };
+            description = ''
+              Configuration for Nixpkgs evaluations.
+            '';
+            type = types.lazyAttrsOf (submoduleWithModules { } [
+              ({ config, name, options, ... }: {
+                _file = ./nixpkgs.nix;
+                options = {
+                  output = mkOption {
+                    default = rootConfig.nixpkgs.evals.${name}.input config.settings;
+                    defaultText = literalExpression
+                      ''config.nixpkgs.evals.''${name}.input config.perSystem.nixpkgs.''${name}.settings'';
+                    description = mdDoc ''
+                      Evaluated Nixpkgs.
+                    '';
+                    type = types.raw;
+                  };
+                  settings = mkOption {
+                    default = { };
+                    description = mdDoc ''
+                      Settings argument for the Nixpkgs evaluations of {option}`perSystem.nixpkgs.evals.<name>`.
+                    '';
+                    type = (getOptionSubOptions [ name ] rootOptions.nixpkgs.evals).settings.type;
+                  };
+                };
+                config = {
+                  settings = mkDefault rootConfig.nixpkgs.evals.${name}.settings;
+                };
+              })
+              # Separate module, for type merging 
+              ({ config, name, options, ... }: {
+                _file = ./nixpkgs.nix;
+                options = {
+                  # mkSubmoduleOptions can't be used here due to `shorthandOnlyDefinesConfig`.
+                  settings = mkSubmoduleOptionsWithShorthand {
+                    crossSystem = mkOption {
```

Comment:
I would recommend to switch to the new `hostPlatform` + optional `buildPlatform` combination that NixOS provides. I haven't got around to propagating that to the Nixpkgs function arguments, but the user interface here would already benefit from it.

Reply:
Sure thing. I forgot that a `system` double is exposed by `platform.system` too. I'm imagining it's okay to `apply = lib.systems.elaborate;` here, as with NixOS `nixpkgs.hostPlatform`?